### PR TITLE
🦋 Buyer accounts edit

### PR DESCRIPTION
**Pull request by josemigallas**
[THREESCALE-10449: Audience > Accounts > Edit](https://issues.redhat.com/browse/THREESCALE-10449)

Before:
Screenshot 2023-11-17 at 09 48 43

After:
Screenshot 2023-11-17 at 09 48 28

### CODE REVIEW

**Comment by mayorova** [comment]
IMO, the "Cancel" link-style button looks weird between the other two buttons... 
I'd rather put it on the right, and the Delete - in the middle.
But maybe there are some PF guidelines that say that's not right?

File: app/views/buyers/accounts/edit.html.slim [hunk 1]
```diff
@@ -3,15 +3,14 @@
...
+    = semantic_form_for @account, builder: Fields::PatternflyFormBuilder,
+                        url: admin_buyers_account_path(@account),
+                        html: { class: 'pf-c-form pf-m-limit-width' } do |form|
       = form.user_defined_form
 
-    / TODO: remove div and move class to form element
-    div class='pf-c-form'
       = form.actions do
         button class="pf-c-button pf-m-primary" type="submit" Update Account
+        a class="pf-c-button pf-m-link" href=admin_buyers_account_path(@account) Cancel
```
Comment (lvillen) [inline app/views/buyers/accounts/edit.html.slim:15]:
```suggestion
        a class="pf-c-button pf-m-secondary" href=admin_buyers_account_path(@account) Cancel
```
WDYT about using `secondary` class so it looks more buttonish than it does now?

**Review by lvillen — COMMENTED** [review]
I agree with @mayorova that it looks nicer with the "Cancel" button to the right, but I'd also make it more buttonish as suggested.

File: app/views/buyers/accounts/edit.html.slim (hunk 1 above)
Comment (mayorova) [reply app/views/buyers/accounts/edit.html.slim:15]: > WDYT about using `secondary` class so it looks more buttonish than it does now?

I think this is against PF4 guidelines:

https://v4-archive.patternfly.org/v4/components/button/design-guidelines#usage

> The Cancel button should always be a [link button](https://v4-archive.patternfly.org/v4/components/button/design-guidelines#link-button).


And it's like that on all other forms.

So, I just propose to swap Delete and Cancel.

Comment (josemigallas) [reply app/views/buyers/accounts/edit.html.slim:15]: I confirm making Cancel a secondary is a no go. Also having a Delete button is against the rules, but we lack of a proper design for it. What I suggest is aligning Delete to the right end, moving it out of the "form buttons" group:
Screenshot 2023-11-21 at 13 38 28

Comment (josemigallas) [reply app/views/buyers/accounts/edit.html.slim:15]: Alternatively, we can add a "Danger zone" form group with a link to delete de account.

Comment (lvillen) [reply app/views/buyers/accounts/edit.html.slim:15]: I like the idea of adding a danger zone in the show page, it makes sense to me.

Comment (josemigallas) [reply app/views/buyers/accounts/edit.html.slim:15]: OK that's a substantial change and will require UX and probably Andreu. It also will affect all other delete buttons in edit pages. We can bring it to then next time. In the meantime I suggest we merge this and carry on.

Comment (mayorova) [reply app/views/buyers/accounts/edit.html.slim:15]: @josemigallas is there a problem in having it like this?

image

To me a Cancel link button separating two "normal" buttons looks really weird :shrug:

Comment (josemigallas) [reply app/views/buyers/accounts/edit.html.slim]: OK I am going to revert the cancel button and leave the minimal change. This PR is only about Patternfly. I [created a JIRA](https://issues.redhat.com/browse/THREESCALE-10517) for UX to work on an actual new design of this "danger zone" and additionally we can add it to the tables as a row action.

That said, I think we can merge this and move on. Thanks for your comments and suggestions!!

**Review by lvillen — APPROVED** [review]

**Pull request merged**